parcels/particlefile/baseparticlefile.py

```diff
@@ -213,15 +219,19 @@ def _create_trajectory_records(self, coords):
             self.z.units = "m"
             self.z.positive = "down"
 
-        for vname in self.var_names:
+        for vname, dtype in zip(self.var_names, self.var_dtypes):
             if vname not in self._reserved_var_names():
-                setattr(self, vname, self.dataset.createVariable(vname, "f4", coords, fill_value=np.nan))
+                fill_value = np.nan if dtype[0] == 'f' else np.iinfo(np.dtype(dtype)).max
+                nc_dtype_fmt = dtype[0] + str(int(int(dtype[-2:])/8))
```

Comment:
This could be nicer with a dictionary? Something like declaring at the top of the function
```python
fmt_map = {'float32': 'f4', 'float64': 'f8', 'int32': 'i4', 'int64', 'i8'}
fill_value_map = {'float32': np.nan, 'float64': np.nan, 'int32': np.iinfo(np.int32).max, 'int64': np.iinfo(np.int64).max}
```
And then 
```suggestion
                fill_value = fill_value_map[str(dtype)]
                nc_dtype_fmt = fmt_map[str(dtype)]
```

Reply:
Great idea - have implemented this (along with using the dtype object rather than dtype name, you're right that there's no good justification for that)